var/spack/repos/builtin/packages/dav-sdk/package.py

```diff
@@ -0,0 +1,177 @@
+
+        for variant in _when_variants:
+            conflicts("~" + variant, when="^" + spec)
+
+    # Skip if there is nothing to propagate
+    if not propagate:
+        return
+
+    # Map the propagated variants to the dependency variant.  Some packages may need
+    # overrides to propagate a dependency as something else, e.g., {"visit": "libsim"}.
+    # Most call-sites will just use a list.
+    if not type(propagate) is dict:
+        propagate = dict([(v, v) for v in propagate])
+
+    # Determine the base variant
+    base_variant = ""
+    if when:
+        base_variant = when
+
+    def is_boolean(variant):
+        return "=" not in variant
+
+    # Propagate variants to dependecy
+    for v_when, v_then in propagate.items():
+        if is_boolean(v_when):
+            depends_on(
+                "{0} +{1}".format(spec, v_then), when="{0} +{1}".format(base_variant, v_when)
+            )
+            depends_on(
+                "{0} ~{1}".format(spec, v_then), when="{0} ~{1}".format(base_variant, v_when)
+            )
+        else:
+            depends_on("{0} {1}".format(spec, v_then), when="{0} {1}".format(base_variant, v_when))
+
+
+def exclude_variants(variants, exclude):
+    return [variant for variant in variants if variant not in exclude]
+
+
+class DavSdk(BundlePackage, CudaPackage, ROCmPackage):
+    """Data & Vis SDK
+
+    ** This package is a post-ECP port from the ECP Data & Vis SDK **
+    """
+
+    homepage = "https://dav-sdk.github.io/"
+
+    tags = ["peso", "oasis", "sdk"]
+    maintainers("kwryankrattiger", "vicentebolea")
+
+    version("1.0")
+
+    ############################################################
+    # Variants
+    ############################################################
+
+    # Data
+    variant("adios2", default=False, description="Enable ADIOS2")
+    variant("hdf5", default=False, description="Enable HDF5")
+    variant("pnetcdf", default=False, description="Enable PNetCDF")
+    variant("diy", default=False, description="Enable DIY")
+
+    # Vis
+    variant("ascent", default=False, description="Enable Ascent")
+    variant("paraview", default=False, description="Enable ParaView")
+    variant("visit", default=False, description="Enable VisIt")
+    variant("vtkm", default=False, description="Enable VTK-m")
+    variant("zfp", default=False, description="Enable ZFP")
+
+    # Language Options
+    variant("fortran", default=True, sticky=True, description="Enable fortran language features.")
+
+    ############################################################
+    # Dependencies
+    ############################################################
+    cuda_arch_variants = ["cuda_arch={0}".format(x) for x in CudaPackage.cuda_arch_values]
+    amdgpu_target_variants = ["amdgpu_target={0}".format(x) for x in ROCmPackage.amdgpu_targets]
+
+    dav_sdk_depends_on(
+        "adios2+shared+mpi+python+sst+dataman",
```

Comment:
sst is default in adios2

Reply:
We try to explicitly specify as many variants as makes sense in the SDK to ensure we get a consistent set of packages. We always want SST for adios2 so we duplicated the default variant here.